### PR TITLE
[SYCL-MLIR] Fix test suite buildbots

### PR DESCRIPTION
**Pull request by whitneywhtsang**
1. Remove run line from single_task.cpp
    - `single_task.cpp` is added in `llvm_test_suite`, so no need to execute it here.
2. Add cgeist to sycl-compiler target
    - cgeist needs to be added in the `install` directory for test suite to use it.
3. Do not run test suite for ESIMD, HIP and CUDA
    - `sycl-mlir` branch of `llvm_test_suite` only executes one test case (single_task.cpp), which is not supported for ESIMD, HIP and CUDA.
4. Make `single_task.cpp` consistent between `intel/llvm` and `intel/llvm-test-suite`.

Signed-off-by: Tsang, Whitney <whitney.tsang@intel.com>

### CODE REVIEW

**Comment by Naghasan** [comment]
Why not use https://github.com/intel/llvm-test-suite/tree/intel/SYCL for the test ? We could probably prevent exec of other test for now but that would avoid worrying about the exec environment

**Comment by whitneywhtsang** [comment]
> Why not use https://github.com/intel/llvm-test-suite/tree/intel/SYCL for the test ? We could probably prevent exec of other test for now but that would avoid worrying about the exec environment

We thought it would be easy to have lit in `intel/llvm` repo to run on device, before expanding to test suite. After understanding the changes needed for it to work, it makes sense to revisit using the test suite.

File: polygeist/tools/cgeist/Test/Verification/sycl/single_task.cpp [hunk 1]
```diff
@@ -2,10 +2,6 @@
```
Comment (sommerlukas) [inline polygeist/tools/cgeist/Test/Verification/sycl/single_task.cpp:2]: Do we really need `clang-format off` here? It seems that the `.clang-format` [file](https://github.com/intel/llvm/blob/f26449e6bff15100773151854f143d7b968067df/polygeist/tools/cgeist/Test/.clang-format) in the parent directory already disables clang-format for this directory. 

Also, it seems that the test case is missing the LLVM copyright header, the other tests in the same directory carry the header.

File: polygeist/tools/cgeist/Test/Verification/sycl/single_task.cpp [hunk 2]
```diff
@@ -25,31 +21,20 @@
 // Test that all referenced sycl header functions are generated.
```
Comment (sommerlukas) [inline polygeist/tools/cgeist/Test/Verification/sycl/single_task.cpp:21]: Is this comment accurate? It seems to test that *no* functions are generated.

**Review by sommerlukas — CHANGES_REQUESTED** [review]
I agree that tests executing SYCL code/applications should live in the llvm-test-suite, so it's good to remove that RUN line here.

Just a few nits in the test file.

File: polygeist/tools/cgeist/Test/Verification/sycl/single_task.cpp (hunk 2 above)
Comment (whitneywhtsang) [reply polygeist/tools/cgeist/Test/Verification/sycl/single_task.cpp:21]: It is accurate. We want to check that the functions are `define` and not `declare`.

File: polygeist/tools/cgeist/Test/Verification/sycl/single_task.cpp (hunk 1 above)
Comment (etiotto) [reply polygeist/tools/cgeist/Test/Verification/sycl/single_task.cpp:2]: In vscode if I change a lit test case (.cpp) my editor reformats it on save (And it fails to read the .clang-format file somehow) so I would like to have the `clang-format-off`.

**Review by etiotto — APPROVED** [review]

File: polygeist/tools/cgeist/Test/Verification/sycl/single_task.cpp (hunk 1 above)
Comment (whitneywhtsang) [reply polygeist/tools/cgeist/Test/Verification/sycl/single_task.cpp:2]: > Also, it seems that the test case is missing the LLVM copyright header, the other tests in the same directory carry the header.

All lit tests in `llvm/polygeist/tools/cgeist/Test/Verification` don't have the LLVM copyright, except the ones in `llvm/polygeist/tools/cgeist/Test/Verification/sycl`.

File: polygeist/tools/cgeist/Test/Verification/sycl/single_task.cpp (hunk 2 above)
Comment (sommerlukas) [reply polygeist/tools/cgeist/Test/Verification/sycl/single_task.cpp:21]: Ok, makes sense, thanks for the explanation!

File: polygeist/tools/cgeist/Test/Verification/sycl/single_task.cpp (hunk 1 above)
Comment (sommerlukas) [reply polygeist/tools/cgeist/Test/Verification/sycl/single_task.cpp:2]: > In vscode if I change a lit test case (.cpp) my editor reformats it on save (And it fails to read the .clang-format file somehow) so I would like to have the `clang-format-off`.

Seems like a different extension for the editor solves @etiotto issue, so maybe we can still remove it from the individual test-file here.

Comment (sommerlukas) [reply polygeist/tools/cgeist/Test/Verification/sycl/single_task.cpp]: > All lit tests in `llvm/polygeist/tools/cgeist/Test/Verification` don't have the LLVM copyright, except the ones in `llvm/polygeist/tools/cgeist/Test/Verification/sycl`. Probably makes more sense to remove the LLVM copyright from the `sycl` directory.

Agreed, I think tests in the LLVM repository tend to not have copyright header, so it should be fine to remove them (optionally in this PR, or at a different time).

**Review by sommerlukas — APPROVED** [review]
LGTM.

**Pull request merged**